tests/integration/services/modal-test.js

```diff
@@ -230,26 +230,44 @@ module('Integration | Service | modal', (hooks) => {
 		assert.equal($element.length, 0, 'Modal is removed from DOM');
 	});
 
-	test('it triggers events when a modal is open/closed', (assert) => {
-		const done = assert.async();
+	test('it triggers events when a modal is open/closed', async(assert) => {
```

Comment:
You should create a new test instead of overriding an existing test, I think it does to much things. Try to test only the resolution of the modal:

```js
cases([
  { title: 'resolve' }, 
  { title: 'reject' }
])test('it fulfills promise modal is removed from DOM ', async({ title }, }assert) => {
  cont promise = this.modal.open...

  modal[title]();
  try {
    await modal;
  } catch {}

  assert.dom('[data-id="modalFoo"]').doesNotExists();
});